validators/workingTimelog.form.go

```diff
@@ -0,0 +1,154 @@
+package validators
+
+import (
+	"aio-server/exceptions"
+	"aio-server/gql/inputs/insightInputs"
+	"aio-server/models"
+	"aio-server/pkg/constants"
+	"aio-server/pkg/helpers"
+	"aio-server/repository"
+)
+
+type WorkingTimelogForm struct {
+	Form
+	insightInputs.SelfWorkingTimelogFormInput
+	User           *models.User
+	Issue          *models.Issue
+	Repo           *repository.WorkingTimelogRepository
+	WorkingTimelog *models.WorkingTimelog
+}
+
+func NewWorkingTimelogFormValidator(
+	input *insightInputs.SelfWorkingTimelogFormInput,
+	user *models.User,
+	repo *repository.WorkingTimelogRepository,
+	workingTimelog *models.WorkingTimelog,
+	issue *models.Issue,
+) WorkingTimelogForm {
+	form := WorkingTimelogForm{
+		Form:                        Form{},
+		SelfWorkingTimelogFormInput: *input,
+		User:                        user,
+		Repo:                        repo,
+		WorkingTimelog:              workingTimelog,
+		Issue:                       issue,
+	}
+
+	form.assignAttributes()
+
+	return form
+}
+
+func (form *WorkingTimelogForm) assignAttributes() {
+
+	// regexpHour := `/\d+h/`
+	// regexpMinute := `/\d+m/`
+
+	form.AddAttributes(
+		&StringAttribute{
+			FieldAttribute: FieldAttribute{
+				Code: "description",
+			},
+			Value: helpers.GetStringOrDefault(form.Description),
+		},
+		&TimeAttribute{
+			FieldAttribute: FieldAttribute{
+				Code: "loggedAt",
+			},
+			Value: helpers.GetStringOrDefault(form.LoggedAt),
+		},
+		&IntAttribute[int32]{
+			FieldAttribute: FieldAttribute{
+				Code: "minutes",
+			},
+			Value: helpers.GetInt32OrDefault(form.Minutes),
+		},
+	)
+}
+
+func (form *WorkingTimelogForm) validate() error {
+	form.validateDescription().validateLoggedAt().validateTimes().assignIssueInfo().summaryErrors()
+
+	if form.Errors != nil {
+		return exceptions.NewUnprocessableContentError("", form.Errors)
+	}
+
+	return nil
+}
+
+func (form *WorkingTimelogForm) validateDescription() *WorkingTimelogForm {
+	field := form.FindAttrByCode("description")
+
+	field.ValidateMax(interface{}(int64(constants.MaxLongTextLength)))
+
+	if field.IsClean() {
+		form.WorkingTimelog.Description = form.Description
+	}
+
+	return form
+}
+
+func (form *WorkingTimelogForm) validateLoggedAt() *WorkingTimelogForm {
+	field := form.FindAttrByCode("loggedAt")
+
+	field.ValidateRequired()
+	field.ValidateFormat("2006-01-02", "%y-%m-%d")
+
+	if field.IsClean() {
+		form.WorkingTimelog.LoggedAt = *field.Time()
+	}
+
+	return form
+}
+
+func (form *WorkingTimelogForm) validateTimes() *WorkingTimelogForm {
+	field := form.FindAttrByCode("minutes")
+
+	field.ValidateRequired()
```

Comment:
sao khi required thì trước tiên là phải validate format theo cái regex em đã đề ra chứ nhỉ ?

chứ nếu chỉ như này thì chỉ cho nhập theo `minutes` à

Reply:
cái này do e đổi sang int hoàn toàn nên ko cần format regex gì nữa ạ